### PR TITLE
AD Support (Contd.)

### PR DESCRIPTION
**Pull request by kanav99**
@YingboMa

### CODE REVIEW

**Comment by YingboMa** [comment]
This isn't the correct fix. It's go with Chris' idea of using a hybrid approach of using Falsi and Bisection together.

File: src/scalar.jl [hunk 1]
```diff
@@ -21,26 +21,35 @@ end
 
 function scalar_nlsolve_ad(prob, alg, args...; kwargs...)
   f = prob.f
-  p = value(prob.p)
-  u0 = value(prob.u0)
+  p = value.(prob.p)
```
Comment (YingboMa) [inline src/scalar.jl:24]: dot is not necessary here. See https://github.com/JuliaComputing/NonlinearSolve.jl/blob/master/src/utils.jl#L238-L240

Comment (kanav99) [reply src/scalar.jl:24]: I think it doesn't handle the subtypes of AbstractArray well, it should be -
```julia
value(x::<:AbstractArray{<:Dual}) = map(ForwardDiff.value, x)
```
Am I right?

Comment (kanav99) [reply src/scalar.jl:24]: Sorry my bad, that isn't even right syntax

**Pull request closed without merging**